packages/block-editor/src/components/block-list/block.native.js

```diff
@@ -113,21 +112,19 @@ class BlockListBlock extends Component {
 				accessibilityRole={ 'button' }
 			>
 				<View accessibilityLabel={ accessibilityLabel }>
-					{ showFloatingToolbar && (
-						<FloatingToolbar>
-							{ hasParent && (
-								<Toolbar passedStyle={ styles.toolbar }>
-									<ToolbarButton
-										title={ __( 'Navigate Up' ) }
-										onClick={ () => onSelect( parentId ) }
-										icon={ NavigateUpSVG }
-									/>
-									<View style={ styles.pipe } />
-								</Toolbar>
-							) }
-							<Breadcrumbs clientId={ clientId } />
-						</FloatingToolbar>
-					) }
+					<FloatingToolbar clientId={ clientId }>
+						{ hasParent && (
+							<Toolbar passedStyle={ styles.toolbar }>
+								<ToolbarButton
+									title={ __( 'Navigate Up' ) }
+									onClick={ () => onSelect( parentId ) }
+									icon={ NavigateUpSVG }
+								/>
+								<View style={ styles.pipe } />
+							</Toolbar>
+						) }
+						<Breadcrumbs clientId={ clientId } />
```

Comment:
with slot-fill architecture on `FloatingToolbar` we can also move all the content to FloatingToolbar file and use it here as below:

`<FloatingToolbar clientId={ clientId } />`

Does it make sense or we would like to keep current PR version?

Reply:
@dratwas I refactor to above according to our conversation